models/Asset/Thumbnail/ImageThumbnailTrait.php

```diff
@@ -336,6 +336,15 @@ protected function convertToWebPath(array $pathReference): ?string
         return $path;
     }
 
+    /**
+     * @param bool $deferred
+     * @return string
+     */
+    public function getFrontendPath(bool $deferred = true): string
+    {
+        return Tool::getHostUrl() . $this->getPath($deferred);
```

Comment:
This is not entirely right, what about remote object storages such as S3? 
Those are handled here: https://github.com/pimcore/pimcore/blob/10.x/models/Asset/Thumbnail/ImageThumbnailTrait.php#L314